src/libAtomVM/refc_binary.c

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright (c) 2021 dushin.net
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <stdlib.h>
+
+#include "context.h"
+#include "refc_binary.h"
+// #include "debug.h"
+#include "dictionary.h"
+#include "memory.h"
+#include "tempstack.h"
+
+//#define ENABLE_TRACE
+
+#include "trace.h"
+
+
+struct RefcBinary *refc_binary_create_refc(size_t size)
+{
+    size_t n = sizeof(struct RefcBinary) + size;
+    struct RefcBinary *refc = malloc(n);
+    if (IS_NULL_PTR(refc)) {
+        return NULL;
+    }
+    list_init((struct ListHead *) refc);
```

Comment:
This stops working on memory layout change, use instead `list_init(&refc->head)`